http/controller/baseController.go

```diff
@@ -0,0 +1,10 @@
+package controller
+
+import "net/http"
+
+type CtxKey struct{}
+
+func GetField(r *http.Request, index int) string {
+	fields := r.Context().Value(CtxKey{}).([]string)
+	return fields[index]
```

Comment:
Outro panic aqui esperado para acontecer se o index for maior que o seu slice. Precisa checar o tamanho antes:

```
if index < len(fields) {
    return fields[index]
}
return ""
```